src/api/category/category.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@nestjs/common';
+import { Category, CategorySelect } from './model';
+import { CategoryArgs, CategoryCreateInput } from './dto';
+import { PrismaService } from 'src/shared/datasource/prisma/prisma.service';
+
+@Injectable()
+export class CategoryService {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  public async findOne(
+    { where }: CategoryArgs,
+    { select }: CategorySelect,
+  ): Promise<Category> {
+    return this.prismaService.category.findUnique({
```

Comment:
Once you add the auth check, we will need to take the user id, uuid or email, from the decoded token, instead of receiving it from the input query. (this for all the cases that we need auth) Also, will be nice to add to user type the VENDOR value, so we can have endpoints available only for vendors, like categories, cause a regular user should not be able to add one.